### PR TITLE
Small adjustments + finishing up documentation

### PR DESCRIPTION
**Pull request by tbembersimeao**
- README-developer.md has been completed.
- `examples/protocols_advanced_v0.0.0` was added to include advanced customization examples
- Minor adjustments have also been made, which include:
  - Fixed bulk operation modal when there are multiple bulk operations
  - Added "default" property setting (for setting up default values)
  - Improvements on EntityQuery class (allows multiple ORDER BY statements)

### CODE REVIEW

File: classes/EntityQuery.php [hunk 1]
```diff
@@ -140,4 +146,16 @@ function execute($load_objects = true, $require_all_conds = true) {
...
     }
+
+    protected function _checkAlias($alias, $msg = 'Invalid alias.') {
+        if (preg_match('/[^a-z0-9_]+/', $alias)) {
+            throw new Exception($msg);
+        }
+    }
+
+    protected function _checkField($field) {
+        if (preg_match('/[^a-z0-9\._]+/', $field)) {
```
Comment (pbchase) [inline classes/EntityQuery.php]: I'm not sure what you were getting at here. Testing this against project_ownership 2.0.2, I got this exception:

```
REDCap crashed due to an unexpected PHP fatal error!

Error message: Uncaught Exception: Invalid field. in /vagrant/modules/redcap_entity_v0.0.0/classes/EntityQuery.php:158 Stack trace: #0 /vagrant/modules/redcap_entity_v0.0.0/classes/EntityQuery.php(34): REDCapEntity\EntityQuery->_checkField('CONCAT(p.projec...') #1 /vagrant/modules/redcap_entity_v0.0.0/classes/EntityList.php(669): REDCapEntity\EntityQuery->condition('CONCAT(p.projec...', '%carol%', 'LIKE') #2 /vagrant/modules/project_ownership_v0.0.0/classes/entity/list/ProjectOwnershipList.php(113): REDCapEntity\EntityList->getQuery() #3 /vagrant/modules/redcap_entity_v0.0.0/classes/EntityList.php(365): ProjectOwnership\Entity\ProjectOwnershipList->getQuery() #4 /vagrant/modules/redcap_entity_v0.0.0/classes/EntityList.php(277): REDCapEntity\EntityList->buildTableRows() #5 /vagrant/modules/redcap_entity_v0.0.0/classes/EntityList.php(132): REDCapEntity\EntityList->renderTable() #6 /vagrant/modules/redcap_entity_v0.0.0/classes/Page.php(64): REDCapEntity\EntityList->renderPageBody() #7 /vagrant/modules/redcap_entity_v0.0.0/clas
File: /vagrant/modules/redcap_entity_v0.0.0/classes/EntityQuery.php
Line: 158
```

Are you checking for invalid characters in the `$field`? I suppose the same would apply for _checkAlias on line 151.

**Comment by pbchase** [comment]
Also check the PR I sent against your branch. https://github.com/tbembersimeao/redcap_entity/pull/2

File: classes/EntityQuery.php (hunk 1 above)
Comment (tbembersimeao) [reply classes/EntityQuery.php]: Nice catch @pbchase, I've just removed `_checkField()` method.

**Comment by tbembersimeao** [comment]
@pbchase thanks for the fixes! I've merged https://github.com/tbembersimeao/redcap_entity/pull/2.

**Comment by pbchase** [comment]
Great docs! Thanks so much.

**Pull request merged**